src/vs/workbench/contrib/testing/common/testService.ts

```diff
@@ -156,12 +157,30 @@ export const getAllTestsInHierarchy = async (collection: IMainThreadTestCollecti
 	]).finally(() => l?.dispose());
 };
 
+/**
+ * Waits for the test to no longer be in the "busy" state.
+ */
+export const waitForTestToBeIdle = (testService: ITestService, test: IncrementalTestCollectionItem) => {
+	if (!test.item.busy) {
+		return;
+	}
+
+	return new Promise<void>(resolve => {
```

Comment:
This could be more efficient, but since it'll be used rarely and never in a very hot path (we'll only be waiting on a single test at any given time, and only when the user runs a command) I think this is fine.